### PR TITLE
[BACKLOG-31795] Layout Editor CRUDL

### PR DESCRIPTION
**Pull request by dcleao**
@pentaho/millenniumfalcon please review.

Merge with: https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1486

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 1m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl user-console
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java [hunk 1]
```diff
@@ -118,55 +130,111 @@ public void fileSelected( final RepositoryFile file, String filePath, String fil
...
+                // Save as, overwriting existing file.
+
                 if ( fileExtension != null && tabName.endsWith( fileExtension ) ) {
                   tabName = tabName.substring( 0, tabName.lastIndexOf( fileExtension ) );
                 }
+
+                final CommandResultCallback composedCallback = composeCallbacks( new CommandResultCallback() {
+                  @Override
+                  public void onSuccess() {
+                    Window.setTitle( Messages.getString( "productName" ) + " - " + name ); //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (davidmsantos90) [inline user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java]: //$NON-NLS-1$ //$NON-NLS-2$ comments can be removed, they only served to suppress warning in Eclipse

File: user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java [hunk 2]
```diff
@@ -118,55 +130,111 @@ public void fileSelected( final RepositoryFile file, String filePath, String fil
...
 
-            doSaveAs( navigatorPerspective.getContentTabPanel().getCurrentFrameElementId(), name, path, type, true );
-            Window.setTitle( Messages.getString( "productName" ) + " - " + name ); //$NON-NLS-1$ //$NON-NLS-2$
-            persistFileInfoInFrame();
-            // navigatorPerspective.addRecent(fullPathWithName, name);
-            clearValues();
+            final CommandResultCallback composedCallback = composeCallbacks(new CommandResultCallback() {
+              @Override
+              public void onSuccess() {
+                Window.setTitle( Messages.getString( "productName" ) + " - " + name ); //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (davidmsantos90) [inline user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java]: same here

File: user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java (hunk 1 above)
Comment (dcleao) [reply user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java]: I made special effort of not touching more lines than needed. This is old code. It's probably easier to make a global find/replace in a dedicated CLEANUP PR.

**Comment by dcleao** [comment]
@davidmsantos90 We never catch you off guard :-)

File: user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java (hunk 1 above)
Comment (dcleao) [reply user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java]: I can do it, afterwards.

**Comment by buildguy** [comment]
### :x: Build failed in 51s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl user-console
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Pull request merged**